enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.ts

```diff
@@ -22,17 +27,43 @@ export const useInitData = ({
   isInitialized: boolean;
 } => {
   const dispatch = useDispatch();
+
   const [isInitialized, setIsInitialized] = useState(false);
   const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [sessionTokenState, setSessionTokenState] =
+    useState<EmbeddingSessionTokenState | null>(null);
 
   useEffect(() => {
     registerVisualizationsOnce();
   }, []);
 
+  useEffect(() => {
+    if (config.authType === "jwt") {
+      const updateToken = () => {
+        const currentState = store.getState();
+        setSessionTokenState(getSessionTokenState(currentState));
+      };
+
+      const unsubscribe = store.subscribe(updateToken);
+
+      if (config.jwtProviderUri) {
+        dispatch(getOrRefreshSession(config.jwtProviderUri));
+      }
+
+      updateToken();
+
+      return () => unsubscribe();
+    }
+  }, [config, dispatch]);
```

Comment:
just to be sure, I think `config` could be changed between renders since it's an object. It's safer to specify each key individually.

```suggestion
  }, [config.authType, config.jwtProviderUri, dispatch]);
```

Reply:
the problem is that it causes an annoying type error because the SdkConfigType causes an error to the tune of
```
config.jwtProviderUri doesn't exist when config.authType === "apiKey"
```

How do you think we could solve it? We could change the SdkConfigType to something like:
```
export type SDKConfigType = {
  metabaseInstanceUrl: string;
  font?: string;
  authType: "apiKey" | "jwt";
  apiKey?: string;
  jwtProviderUri?: string;
};
```

as opposed to the current type of 
```
type SDKAuthType =
  | {
      authType: "apiKey";
      apiKey: string;
    }
  | {
      authType: "jwt";
      jwtProviderUri: string;
    };

export type SDKConfigType = {
  metabaseInstanceUrl: string;
  font?: string;
} & SDKAuthType;
```

Reply:
let me try something out

Reply:
This looks a little off, but to accommodate both:
1. rule of hook ESLint rule and
2. the expressiveness of union type.
```ts
  const jwtProviderUri = config.authType === "jwt" ? config.jwtProviderUri : null;
  useEffect(() => {
    if (config.authType === "jwt") {
      const updateToken = () => {
        const currentState = store.getState();
        setSessionTokenState(getSessionTokenState(currentState));
      };

      const unsubscribe = store.subscribe(updateToken);

      if (jwtProviderUri) {
        dispatch(getOrRefreshSession(jwtProviderUri));
      }

      updateToken();

      return () => unsubscribe();
    }
  }, [config.authType, dispatch, jwtProviderUri]);
```

What do you think?

Reply:
looks good to me 😄